### PR TITLE
Fix #2107: Set PropagationPolicy to Background by default

### LINKED ISSUE #2107: Confusing message from kubernetes-client: object is being deleted: secrets "XXXX" already exists
During studying this https://github.com/eclipse/che/issues/16512 that might be related with upgrade of  kubernetes-client  from 4.1.0 to 4.9.0  https://github.com/eclipse/che-parent/pull/159 https://github.com/eclipse/che/pull/16345 we noticed very strange error.
```
aused by: io.fabric8.kubernetes.client.KubernetesClientException: Failure executing: POST at: https://172.30.0.1/api/v1/namespaces/skabashn/secrets. Message: object is being deleted: secrets "workspaceyy5nesnxw954tsbz-sshprivatekeys" already exists. Received status: Status(apiVersion=v1, code=409, details=StatusDetails(causes=[], group=null, kind=secrets, name=workspaceyy5nesnxw954tsbz-sshprivatekeys, retryAfterSeconds=null, uid=null, additionalProperties={}), kind=Status, message=object is being deleted: secrets "workspaceyy5nesnxw954tsbz-sshprivatekeys" already exists, metadata=ListMeta(_continue=null, remainingItemCount=null, resourceVersion=null, selfLink=null, additionalProperties={}), reason=AlreadyExists, status=Failure, additionalProperties={}).
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.requestFailure(OperationSupport.java:568)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.assertResponseCode(OperationSupport.java:507)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:471)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleResponse(OperationSupport.java:430)
	at io.fabric8.kubernetes.client.dsl.base.OperationSupport.handleCreate(OperationSupport.java:251)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.handleCreate(BaseOperation.java:815)
	at io.fabric8.kubernetes.client.dsl.base.BaseOperation.create(BaseOperation.java:333)
	at org.eclipse.che.workspace.infrastructure.kubernetes.namespace.KubernetesSecrets.create(KubernetesSecrets.java:49)
	... 10 common frames omitted
```
Any ideas?

### PR DESCRIPTION
**Pull request by rohanKanojia**
Fix #2107

This should also take care of #2147

### CODE REVIEW

**Comment by rohanKanojia** [comment]
retest this please

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/PropagationPolicyConfigurable.java [hunk 1]
```diff
@@ -17,5 +17,17 @@
...
+   * Either this field or OrphanDependents may be set, but not both.
+   * The default policy is decided by the existing finalizer set in the metadata.finalizers and the resource-specific default policy.
+   * Acceptable values are:
+   *    'Orphan' - orphan the dependents;
+   *    'Background' - allow the garbage collector to delete the dependents in the background;
+   *    'Foreground' - a cascading policy that deletes all dependents in the foreground.
+   *
+   * @param propagationPolicy propagation policy in form of string
+   * @return resource
+   */
```
Comment (manusa) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/PropagationPolicyConfigurable.java:31]: This JavaDoc will be scrambled when processed (no line breaks), try any of the proposed alternatives.
```suggestion
  /**
   * Whether and how garbage collection will be performed.
   * Either this field or OrphanDependents may be set, but not both.
   *
   * <p>The default policy is decided by the existing finalizer set in the metadata.finalizers and the resource-specific default policy.
   * 
   * <p>Acceptable values are:
   * <ul>
   * <li>'Orphan' - orphan the dependents;</li>
   * <li>'Background' - allow the garbage collector to delete the dependents in the background;</li>
   * <li>'Foreground' - a cascading policy that deletes all dependents in the foreground.</li>
   * </ul>
   *
   * @param propagationPolicy propagation policy in form of string
   * @return resource
   */
```
Or:
```suggestion
  /**
   * Whether and how garbage collection will be performed.
   * Either this field or OrphanDependents may be set, but not both.
   *
   * <p>The default policy is decided by the existing finalizer set in the metadata.finalizers and the resource-specific default policy.
   *
   * <p>Acceptable values are:
   * <br>Orphan' - orphan the dependents;</li>
   * <br>Background' - allow the garbage collector to delete the dependents in the background;</li>
   * <br>Foreground' - a cascading policy that deletes all dependents in the foreground.</li>
   *
   * @param propagationPolicy propagation policy in form of string
   * @return resource
   */
```

**Comment by manusa** [comment]
If propagation policy is a known set of values, why not use an enum instead?

Only advantage of current implementation (free String) is that in the future k8s adds other valid values. But in that case, maybe some warning should be issued if the propagation policy is not within the accepted set, or at least create some publicly accessible constants. 

We want to prevent users who type `withPropagationPolicy("Bakcground")` becoming crazy

File: kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/OperationSupport.java [hunk 2]
```diff
@@ -221,7 +224,10 @@ protected void handleDelete(URL requestUrl, long gracePeriodSeconds, String prop
      * Either the propagation policy or the orphan dependent (deprecated) property must be set, but not both.
      */
     if (propagationPolicy != null) {
-      deleteOptions.setPropagationPolicy(propagationPolicy);
+      if (Arrays.stream(DeletionPropagation.values()).noneMatch(deletionPropagation -> deletionPropagation.name().equals(propagationPolicy))) {
+        logger.warn("Unknown DeletionPropagation policy used for deletion: {}", propagationPolicy);
+      }
```
Comment (manusa) [inline kubernetes-client/src/main/java/io/fabric8/kubernetes/client/dsl/base/OperationSupport.java:229]: This check shouldn't be necessary any more as we are using an enum

**Comment by manusa** [comment]
[merge]

**Pull request merged**